bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/handler/BigAssFanHandler.java

```diff
@@ -1045,26 +1052,28 @@ protected synchronized void connect() {
             }
             logger.trace("Connecting to {} at {}", thing.getUID(), ipAddress);
 
+            Socket localFanSocket = new Socket();
             // Open socket
             try {
-                fanSocket = new Socket();
-                fanSocket.bind(new InetSocketAddress(ifAddress, 0));
-                fanSocket.connect(new InetSocketAddress(ipAddress, BAF_PORT), SOCKET_CONNECT_TIMEOUT);
-            } catch (IOException e) {
-                logger.debug("IOException connecting to  {} at {}: {}", thing.getUID(), ipAddress, e.getMessage());
+                localFanSocket.bind(new InetSocketAddress(ifAddress, 0));
+                localFanSocket.connect(new InetSocketAddress(ipAddress, BAF_PORT), SOCKET_CONNECT_TIMEOUT);
+                fanSocket = localFanSocket;
```

Comment:
In case of exceptions thrown, do you want `fanSocket` to be assigned new `Socket()` or not? Previously it was assigned immediately, also in case of exceptions.

Reply:
Good one!

Reply:
Redundant assignment now as already assigned in line 1053.